### PR TITLE
fargate implementation

### PR DESCRIPTION
**Pull request by lucifer7663**
Fixes #

- [ ] Does not depend on other PRs
- [x] Depends on other PR being merged first: #371 

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
- implemented aws fargate for video compression
## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: src/main/java/com/serverless/Handler.java [hunk 1]
```diff
@@ -47,7 +46,7 @@ public ApiGatewayResponse handleRequest(Map<String, Object> input, Context conte
         AmazonDynamoDB ddb = AmazonDynamoDBClientBuilder.standard()
                 .withCredentials(new DefaultAWSCredentialsProviderChain())
                 .withRegion(Regions.AP_NORTHEAST_1).build();
-        if (input.containsKey("routeKey") && input.get("routeKey").equals("POST /user-activity")
+        if (input.containsKey("routeKey") && input.get("routeKey").equals("POST /insert")
```
Comment (mhazad) [inline src/main/java/com/serverless/Handler.java:49]: "user-activity" was perfect as a resource. Why it is changed to "insert"?

File: src/main/java/com/serverless/Handler.java [hunk 2]
```diff
@@ -81,7 +80,7 @@ public ApiGatewayResponse handleRequest(Map<String, Object> input, Context conte
                         .build();
             }
         }
-        if (input.containsKey("routeKey") && input.get("routeKey").equals("GET /user-activity")
+        if (input.containsKey("routeKey") && input.get("routeKey").equals("GET /query")
```
Comment (mhazad) [inline src/main/java/com/serverless/Handler.java:83]: "user-activity" was perfect as a resource. Why it is changed to "query"?

File: src/main/java/com/serverless/Handler.java [hunk 3]
```diff
@@ -286,13 +285,36 @@ public ApiGatewayResponse handleRequest(Map<String, Object> input, Context conte
...
+                            overrides.withContainerOverrides(containerOverrides);
+
+                            RunTaskRequest request = new RunTaskRequest()
+                                    .withCluster(clusterName)
+                                    .withTaskDefinition(taskDefinition)
+                                    .withOverrides(overrides)
+                                    .withCount(1)
+                                    .withNetworkConfiguration(configuration)
+                                    .withLaunchType(LaunchType.FARGATE);
+                            ecs.runTask(request);
```
Comment (mhazad) [inline src/main/java/com/serverless/Handler.java:317]: Where is your defensive code here? What will happen if this encounters an exception? You made similar mistakes earlier and proudly continuing it. Bravo! 👏

**Review by mhazad — CHANGES_REQUESTED** [review]
You are continuing to make the same mistakes. If you continue to do this then it will be nearly impossible for you to continue with us. Please be cautious with your keyboard and output.

File: src/main/java/com/serverless/Handler.java (hunk 1 above)
Comment (lucifer7663) [reply src/main/java/com/serverless/Handler.java]: resolved the problem

File: src/main/java/com/serverless/Handler.java (hunk 3 above)
Comment (lucifer7663) [reply src/main/java/com/serverless/Handler.java]: introduced defensive code

File: src/main/java/com/serverless/Handler.java (hunk 2 above)
Comment (lucifer7663) [reply src/main/java/com/serverless/Handler.java]: resolve the problem

File: src/main/java/com/serverless/CompressVideo.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+public class CompressVideo {
+    public static void main(String[] args) {
+        String bucketName = System.getenv("BucketName");
+        String fileName = System.getenv("fileName");
+        String compressResult = compressVideos(fileName, bucketName);
```
Comment (mhazad) [inline src/main/java/com/serverless/CompressVideo.java:19]: BucketName is Pascal case and fileName is camel case. Please follow one single convention.

**Review by mhazad — CHANGES_REQUESTED** [review]
Good attempt 👍  Please address the feedback.

**Comment by mhazad** [comment]
One more general feedback 

In the terraform file you have defined the environment variables like this
```tf
environment {
    variables = {
      nayem_bucket_name = var.nayem_bucket_name
      nayem_sns = var.nayem_sns
      nayem_sqs_name = var.nayem_sqs_name
      nayem_ddb_table = var.nayem_ddb_table
      clusterName = var.clusterName
      subnets = var.subnets
      container_definitions_name = var.container_definitions_name
      task_definition_name = var.task_definition_name
      container_image = var.container_image
    }
  }
```

And in the serverless.yml you have defined the environment variables like this

```yml
 environment:
    BucketName: ${env:BUCKETNAME}
    QueueName: ${env:QUEUENAME}
    TableName: ${env:TABLENAME}
    Table_Name: ${env:QUERYTABLENAME}
    TopicName: ${env:TOPICNAME}
    DisplayName: ${env:DISPLAYNAME}
    baseUrl: ${env:BASEURL}
    webhook: ${env:WEBHOOK}

```

Ideally, it should be the same as for the terraform it should be like below and then for this you do not need to change any code in the handler.java

```tf
      BucketName = var.nayem_bucket_name
      TopicName = var.nayem_sns
      QueueName = var.nayem_sqs_name
      TableName= var.nayem_ddb_table
```

File: src/main/java/com/serverless/CompressVideo.java (hunk 4 above)
Comment (lucifer7663) [reply src/main/java/com/serverless/CompressVideo.java]: fileName is now in Pascal Case

**Review by mhazad — APPROVED** [review]
Now its look really good to me! Good Job and keep up the quality.

**Pull request merged**